### DISCUSSION TITLE
Decrypt Adobe.PubSec encrypted PDF

### DISCUSSION THREAD

**Discussion by redfast00** [MatthiasValvekens/pyHanko#410] [Q&A]
From what I can read in the code, there appears to be functionality related to Adobe.PubSec encrypted PDFs. Is it possible to decrypt those using pyHanko? I have a .p12 file, and a password for that certificate.

If this is possible, how do I do this?

**Comment by MatthiasValvekens** [comment]
Hello,


That's indeed possible. To do that using the CLI, run `pyhanko decrypt pkcs12 --help` for more info.

If you're looking to do this using pyHanko as a library: there's no specific written documentation for this procedure (other than the API docs themselves), but if you look around for `decrypt_with_pkcs12` in the codebase, you can probably figure it out from the implementation of the command I referenced above.
